### ISSUE TITLE
-L option when tracing in other languages (e.g. Python)

### ISSUE THREAD

**Issue by henrychoi7** [namhyung/uftrace#926]
Hi there,

I was trying to trace python files in another uftrace directory (I already installed uftrace to the system environment by executing `sudo make install` in master branch and compiled uftrace again from `origin/review/python-tracing-v3` branch). Therefore, I tried with `./uftrace` but nothing happens after executing python files like below.

```
$ ./uftrace --python --no-libcall test.py
1
(nothing happens)
```

Also, if uftrace couldn't find the library, it terminates with a error message "cannot found libmcount.so". I know that uftrace finds `libmcount` in a sequence of `./libmcount/libmcount.so` -> `./libmcount.so` -> `system`, so it terminates when `libpath == NULL`. Eventually, I had to input `-L.` option to use `libmcount` from the uftrace directory, not from the system.

```
$ ./uftrace --python --no-libcall -L. test.py
1
# DURATION     TID     FUNCTION
            [107992] | <module:__main__>() {
            [107992] |   a() {
   0.992 us [107992] |     b();
   4.533 us [107992] |   } /* a */
 213.450 us [107992] |   /* linux:schedule */
 305.949 us [107992] |   /* linux:schedule */
 564.127 us [107992] | } /* <module:__main__> */
  10.061 us [107992] | _remove();
   1.922 us [107992] | _remove();
```

To make this more convenient, I added `lib_path` value in `uftrace.c`:
https://github.com/namhyung/uftrace/blob/f35749873f05e2e8d7fc4499ca45c07ea62ec536/uftrace.c#L1052

and commented this line in `record.c`:
https://github.com/namhyung/uftrace/blob/f35749873f05e2e8d7fc4499ca45c07ea62ec536/cmds/record.c#L130

As a result, I could trace python files without `-L` option in the uftrace directory.

```
$ ./uftrace --python --no-libcall test.py
1
# DURATION     TID     FUNCTION
            [108001] | <module:__main__>() {
            [108001] |   a() {
   0.697 us [108001] |     b();
   3.415 us [108001] |   } /* a */
 518.306 us [108001] |   /* linux:schedule */
 309.994 us [108001] |   /* linux:schedule */
 870.406 us [108001] | } /* <module:__main__> */
   7.563 us [108001] | _remove();
   2.147 us [108001] | _remove();
```

From the user perspective (not for development/test purpose), I think it is more easy to use without confusing the library location and adding '-L' option.

**Comment by namhyung** [comment]
The `-L` option is for debugging/testing and not for users.  After proper installation process, all libraries should be in a system directory and be loaded without a problem IMO.

**Comment by henrychoi7** [comment]
@namhyung Sorry, my explanation wasn't enough. Yes, I didn't knew about the `-L` option until I've found the `libmcount` and analyzed the program reading library path. I didn't download the uftrace source for debugging/testing. I just wanted to **use** it on a other language. Therefore I thought; does it necessary for the users to know about the `-L` option? If then, I thought it would be helpful to have such a document or a feature that users won't have to use the `-L` option (it **doesn't** mean the removal of `-L` option. I know that it is needed for debugging/testing) or a guide for other languages IMO.

**TL;DR**
I didn't knew that the `-L` option is needed to trace python files. Because I was looking from a user perspective.

**Comment by namhyung** [comment]
The python function tracing is still in development.  Once it's done all necessary libraries should be installed in a proper directory so users don't need to use `-L` option to specify some directory.  Before that we are using it just for easier testing.

**Comment by henrychoi7** [comment]
I understand. It was a little rushed to think of this ‘-L’ option. I’ll keep testing and see what I can do for further python tracing support. Thank you for your opinion! :)
